src/extension.ts

```diff
@@ -86,47 +88,90 @@ class MessageHandler implements IMessageHandler {
 	}
 }
 
-export function activate(context: vscode.ExtensionContext) {
-	let godotProjectDir = vscode.workspace.workspaceFolders?.find((workspaceFolder) => {
-		let rootPath = workspaceFolder.uri.fsPath;
-		let godotProjectFile = path.join(rootPath, 'project.godot');
-
-		return fs.existsSync(godotProjectFile);
-	})?.uri.fsPath;
-
-	if (godotProjectDir === undefined) {
+export async function activate(context: vscode.ExtensionContext) {
+	let foundProjects: ProjectLocation[] = await findProjectFiles();
+	// No project.godot files found. The extension doesn't need to do anything more.
+	if (foundProjects.length == 0) {
 		return;
 	}
 
-	godotProjectDir = fixPathForGodot(godotProjectDir);
+	// Setup the status bar / project selector and prompt for project if necessary
+	const commandId = 'csharpGodot.selectProject';
+	statusBarItem = vscode.window.createStatusBarItem(vscode.StatusBarAlignment.Left, 0);
+	statusBarItem.command = commandId;
+	statusBarItem.show();
+	context.subscriptions.push(statusBarItem);
+	context.subscriptions.push(vscode.commands.registerCommand(commandId, async () => {
+		let project = await promptForProject();  // project.godot
+		if (project !== undefined) {
+			setupProject(project, context);
+		}
+	}));
+
+	// One project.godot files found. Use it.
+	if (foundProjects.length == 1) {
+		setupProject(foundProjects[0], context);
+	}
+	// Multiple project.godot files found. Prompt the user for which one they want to use.
+	else {
+		let project = await promptForProject();
+		if (project !== undefined) {
+			setupProject(project, context);
+		}
+	}
+}
 
-	client = new Client('VisualStudioCode', godotProjectDir, new MessageHandler(), new Logger());
+function setupProject(project: ProjectLocation, context: vscode.ExtensionContext) {
+	let statusBarPath:string = project.relativeProjectPath === '.' ? './'  : project.relativeProjectPath;
```

Comment:
A single `.` looks kinda weird when the selected `project.godot` file is in the root workspace directory, so I display `./` in that case to make it more obvious. Absolute paths are an option too but long paths may cause other status bar items to disappear.

Reply:
How about `path.dirname(project.absoluteProjectPath)`?

Reply:
Yeah that could work, i'm just worried that if someone's absolute path is too long it could cause other status bar items to be hidden. Especially on laptops with lowres screens.

Also sidenote; `project.absoluteProjectPath` already is the project directory :slightly_smiling_face: 
```
project.absoluteProjectPath <-- path to directory container project.godot
project.absoluteFilePath <-- path to project.godot itself
```

Reply:
Oh right, I meant `path.basename`

```js
const p = '/path/to/project/project.godot';
path.dirname(p); // /path/to/project
path.basename(path.dirname(p)); // project
```

Reply:
_Personally_ as a user I could see that being ambiguous since the godot project directory might be nested a few levels down in the workspace. A real path of `./a/b/godot` would be displayed as `godot` and it wouldn't be immediately obvious were the project was located. Now that's not likely to happen for most users, but I could see myself ending up with a project setup like that :wink: 

Its up to you fine folks though, if you have a strong preference I'll change it.

Reply:
I don't have a strong preference, personally most of the workspaces I'll open in VSCode will contain only one project, so I'll let @neikeq decide

Reply:
Makes sense. And that's probably true for most people too; they won't even have to interact with this feature or care about the status bar since they only have one Godot project.

Reply:
I prefer the current relative paths and `./`.